### ISSUE TITLE
Introduction of erroneous newlines in Requires-Dist

### ISSUE THREAD

**Issue by pelson** [wimglenn/setuptools-ext#9]
Thanks for this project! Was suggested to me in https://github.com/pypa/setuptools/discussions/4220#discussioncomment-9130979. I noticed that adding a ``Requires-External`` modified other parts of the metadata in an incorrect way...

With the normal setuptools backend, I have a record such as the following in METADATA:

```
Requires-Dist: typing-extensions <5,>=4.2 ; (python_version < "3.8") and extra == 'all'
```

Meanwhile, with the setuptools-ext backend enabled, you get:

```
Requires-Dist: typing-extensions <5,>=4.2 ;
 (python_version < "3.8") and extra == 'all'
```

This is not a valid specification, and results in a unusable wheel. It looks like this comes from email, where https://docs.python.org/3/library/email.generator.html#email.generator.BytesGenerator has a maxlinelength parameter. The value comes from the policy (that is not specified) that you get when you call ``email.message_from_string`` - the limit is 78 chars https://docs.python.org/3/library/email.policy.html#email.policy.Policy.max_line_length.

**Comment by wimglenn** [comment]
@pelson  Thanks for the report. Fixed in [0.6](https://pypi.org/project/setuptools-ext/0.6/#history).

**Comment by pelson** [comment]
Nice. Tested and it works great! Thanks for the project! :+1:
